lib/Rex/Helper/Rexfile/ParamLookup.pm

```diff
@@ -0,0 +1,103 @@
+#
+# (c) Jan Gehring <jan.gehring@gmail.com>
+#
+# vim: set ts=3 sw=3 tw=0:
+# vim: set expandtab:
+
+package Rex::Helper::Rexfile::ParamLookup;
+
+use strict;
+use warnings;
+
+# VERSION
+
+use Devel::Caller;
+use Data::Dumper;
+require Rex::Exporter;
+require Rex::Commands;
+
+use base qw(Rex::Exporter);
+use vars qw (@EXPORT);
+
+@EXPORT = qw(param_lookup);
+
+sub param_lookup {
+  my ( $key, $default ) = @_;
+
+  my $ret;
+
+  my ($caller_pkg) = caller(0);
+
+  my @args = Devel::Caller::caller_args(1);
+  if ( ref $args[0] eq "HASH" ) {
+    if ( exists $args[0]->{$key} ) {
+      $ret = $args[0]->{$key};
+    }
+  }
+
+  if ( !$ret ) {
+
+    # check if cmdb is loaded
+    my ($use_cmdb) = grep { m/CMDB\.pm/ } keys %INC;
+    if ($use_cmdb) {
+
+      # look inside cmdb
+      my $cmdb_key = "${caller_pkg}::$key";
+      $ret = Rex::Commands::get( Rex::CMDB::cmdb($cmdb_key) );
+
+      if ( !$ret ) {
+
+        # check in resource
+        if ($Rex::Resource::INSIDE_RES) {
+          $cmdb_key =
+              "${caller_pkg}::"
+            . $Rex::Resource::CURRENT_RES[-1]->display_name
+            . "::$key";
+          $ret = Rex::Commands::get( Rex::CMDB::cmdb($cmdb_key) );
+        }
+      }
+
+      if ( !$ret ) {
+
+        # check in global namespace
+        $ret = Rex::Commands::get( Rex::CMDB::cmdb($key) );
+      }
+    }
+  }
+
+  if ( !$ret ) {
+    $ret = $default;
+  }
+
+  # make the variable also known to templates
+  Rex::Commands::set( $key => $ret );
+
+  return $ret;
+}
+
+1;
+
+=pod
+
+=head1 NAME
+
+Rex::Helper::Rexfile::ParamLookup - A command to manage task parameters.
+
+A command to manage task parameters. Additionally it register the parameters as template values.
+
+This module also looks inside a CMDB (if present) for a valid key.
+
+
+=head1 SYNOPSIS
+
+ task "setup", make {
```

Comment:
the `make` keyword is the same as `sub`. I'll need to update the docs and change it to sub.
i added `make` in the past for a colleague (kind of a joke), because he was totally in love with "make". And sometimes i use it instead of `sub`